src/main/java/com/ss/training/lms/controller/AdminGenreController.java

```diff
@@ -0,0 +1,131 @@
+package com.ss.training.lms.controller;
+
+import java.sql.SQLException;
+import java.util.List;
+
+import com.ss.training.lms.entity.Genre;
+import com.ss.training.lms.service.admin.AdminGenreService;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author Trevor Huis in 't Veld
+ */
+@RestController
+@RequestMapping("/lms/admin/genre")
+public class AdminGenreController {
+	
+	@Autowired
+	AdminGenreService genreService;
+	
+	/**
+	 * 
+	 * @param genreId
+	 * @return
+	 */
+	@GetMapping(path="/{genreId}")
+	public ResponseEntity<Genre>readGenre(@PathVariable int genreId) {
+		Genre genre = null;
+		HttpStatus status = HttpStatus.OK;
+		try {
+			genre= genreService.readGenre(genreId);
+			if (genre == null) {
+				status = HttpStatus.NOT_FOUND;
+			}
+		} catch (SQLException e) {
+			status = HttpStatus.INTERNAL_SERVER_ERROR;
+		}
+		return new ResponseEntity<Genre>(genre, status);
+	}
+
+	/**
+	 * 
+	 * @return
+	 */
+	@GetMapping()
+	public ResponseEntity<List<Genre>> readGenres() {
+		List<Genre> genres = null;
+		HttpStatus status = HttpStatus.OK;
+        try {
+			genres= genreService.readAllGenres();
+			if (genres == null) {
+				status = HttpStatus.NO_CONTENT;
+			}
+		} catch (ClassNotFoundException | SQLException e) {
+			status = HttpStatus.INTERNAL_SERVER_ERROR;
+		}
+        return new ResponseEntity<List<Genre>>(genres, status);
+	}
+
+	/**
+	 * 
+	 * @param genre
+	 * @return
+	 */
+	@PostMapping()
+	public ResponseEntity<Genre> createGenre(@RequestBody Genre genre) {
+		HttpStatus status = HttpStatus.BAD_REQUEST;
+		if(genre == null || genre.getGenreID() == null || genre.getGenreName().length() > 45)
```

Comment:
It's still possible to check the length of a null string here.

Reply:
geez my bad. I misunderstood your comment.

Reply:
No worries! I could have been more specific.